### PR TITLE
Feature/add legend to quantification

### PR DESCRIPTION
**Pull request by weberjavi**
This PR adds a legend to explain quantification values on GHG emissions chart .  
[Basecamp thread](https://basecamp.com/1756858/projects/13795275/todos/350625874)
[Zeplin designs](https://app.zeplin.io/project/598b1dc89689e51e1cd1b5ce/screen/5af072c057be43bd03af46c0) .  
It also fixes some `z-index` issues on country compare selectors as defined on this [basecamp thread](https://basecamp.com/1756858/projects/13795275/todos/351627392)

Can be tested on [http://localhost:3000/countries/TUR](http://localhost:3000/countries/TUR)
image

### CODE REVIEW

File: app/javascript/app/components/charts/stacked-area/stacked-area-component.jsx [hunk 1]
```diff
@@ -215,9 +217,12 @@ class ChartStackedArea extends PureComponent {
...
                 activePoint &&
                 (point.x === activePoint.x && point.y === activePoint.y);
-              let colorPoint = point.label === 'BAU' ? '#113750' : '#ffc735';
+              let colorPoint =
+                point.label === 'BAU'
+                  ? QUANTIFICATION_COLORS.BAU
+                  : QUANTIFICATION_COLORS.QUANTIFIED;
               if (point.label === 'No quantifiable target') {
-                colorPoint = '#b1b1c1';
+                colorPoint = QUANTIFICATION_COLORS.NOT_QUANTIFICABLE;
```
Comment (simaob) [inline app/javascript/app/components/charts/stacked-area/stacked-area-component.jsx]: maybe: NOT_QUANTIFIABLE

File: app/javascript/app/components/country/country-ghg-emissions/country-ghg-emissions-selectors.js [hunk 2]
```diff
@@ -242,6 +243,29 @@ export const getQuantificationsData = createSelector(
...
+    );
+    const nq = quantifications.find(q => q.label === 'No quantifiable target');
+    if (bau) {
+      config.push(QUANTIFICATIONS_CONFIG.bau);
+    }
+    if (qua) {
+      config.push(QUANTIFICATIONS_CONFIG.quantified);
+    }
+    if (nq) {
+      config.push(QUANTIFICATIONS_CONFIG.not_quantificable);
```
Comment (simaob) [inline app/javascript/app/components/country/country-ghg-emissions/country-ghg-emissions-selectors.js]: rename to: not_quantifiable

File: app/javascript/app/data/constants.js [hunk 3]
```diff
@@ -13,6 +13,24 @@ export const CALCULATION_OPTIONS = {
...
+
+export const QUANTIFICATIONS_CONFIG = {
+  bau: { label: 'Business as usual', color: QUANTIFICATION_COLORS.BAU },
+  quantified: {
+    label: 'Quantified targets',
+    color: QUANTIFICATION_COLORS.QUANTIFIED
+  },
+  not_quantificable: {
+    label: 'Target not quantifiable',
+    color: QUANTIFICATION_COLORS.NOT_QUANTIFICABLE
```
Comment (simaob) [inline app/javascript/app/data/constants.js]: another place.

Comment (weberjavi) [reply app/javascript/app/data/constants.js]: :/ I forgot my dictionary

**Comment by simaob** [comment]
Looks good to me now!

**Pull request merged**